### PR TITLE
get restaurant detailed info through Yelp API

### PR DESCRIPTION
**Pull request by Smao0875**
api/models/Restaurant.js is not neccessary.
Should we remove it or keep it for future use?

### CODE REVIEW

**Comment by Smao0875** [comment]
Yelp API-KEY:
Bearer WdCKkVBICcEcVeO8C96lCEcTTkog2L9cP2VaRSyaaYUck0CdJNguo40wmyCV4pd7sE5knHHuLG6HKejNqRr41oCkxhMU565DV8kYUshzRkcMo8t8aclSwZaCcpE1WnYx

Yelp id:
dentistry-for-kids-and-adults-canyon-country

File: api/controllers/PromotionController.js [hunk 1]
```diff
@@ -1,7 +1,9 @@
 var Promotion = require('../models/Promotion');
+var Restaurant = require('../models/Restaurant');
 var User = require('../models/User');
 var auth = require('../utils/auth');
 var csv = require('fast-csv');
+var superagent = require('superagent');
```
Comment (liptonday) [inline api/controllers/PromotionController.js]: this is a node.js native thing? no need to add to `package.json`?

File: api/controllers/PromotionController.js [hunk 2]
```diff
@@ -35,7 +37,22 @@ var PromotionController = {
...
-    }
+    },
+
+    getRestaurantById: function(req, res) {
+        var id = req.swagger.params.id.value;
+        var token = req.headers.authorization;
+
+        superagent
+        .get('https://api.yelp.com/v3/businesses/'+ id)
+        .set('Authorization', token)
```
Comment (liptonday) [inline api/controllers/PromotionController.js]: this `token` coming from API user is the auth token issued by our api server, not the token by yelp

File: api/models/Restaurant.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
+var mongoose = require('mongoose');
```
Comment (liptonday) [inline api/models/Restaurant.js:3]: nice to have it for reference purpose, can you add some comments on top so anyone comes here they know this is not the database data model?

File: api/controllers/PromotionController.js [hunk 4]
```diff
@@ -1,7 +1,9 @@
 var Promotion = require('../models/Promotion');
+var Restaurant = require('../models/Restaurant');
```
Comment (liptonday) [inline api/controllers/PromotionController.js]: can be removed if not being used

File: api/controllers/PromotionController.js (hunk 1 above)
Comment (Smao0875) [reply api/controllers/PromotionController.js]: I guess we don't need to add it.
-- It works pretty well.

File: api/controllers/PromotionController.js (hunk 2 above)
Comment (Smao0875) [reply api/controllers/PromotionController.js]: Actually it is the api-key I got from Yelp.
capture

File: api/models/Restaurant.js (hunk 3 above)
Comment (Smao0875) [reply api/models/Restaurant.js:3]: Sure, will do.

File: api/controllers/PromotionController.js (hunk 4 above)
Comment (Smao0875) [reply api/controllers/PromotionController.js]: Cool.

File: api/controllers/PromotionController.js (hunk 1 above)
Comment (Smao0875) [reply api/controllers/PromotionController.js]: I will add it to package.json anyway, just in case.

Comment (Smao0875) [reply api/controllers/PromotionController.js]: Here is the screen shot I got when I run 
-- npm install superagent
capture

Seems that it is under supertest and swagger-express-mw.
-- And the package.json file is not changed at all.

File: api/controllers/PromotionController.js (hunk 2 above)
Comment (liptonday) [reply api/controllers/PromotionController.js]: Sorry I don't understand, this `token` is from request header right? And how do you pass Yelp api key to the request header?

Comment (Smao0875) [reply api/controllers/PromotionController.js]: Oh, wait a sec.
I should not put this API-KEY in the request header.
Cliend side should not know it.
It should only be known by server, correct?

Comment (liptonday) [reply api/controllers/PromotionController.js]: ya that's confusing me

Comment (liptonday) [reply api/controllers/PromotionController.js]: you can hardcode it for now, I have a PR coming soon regarding the config

**Review by liptonday — APPROVED** [review]
👍

**Pull request merged**